expression/builtin_other.go

```diff
@@ -75,64 +76,71 @@ type setVarFunctionClass struct {
 	baseFunctionClass
 }
 
-func (c *setVarFunctionClass) getFunction(ctx context.Context, args []Expression) (builtinFunc, error) {
-	err := errors.Trace(c.verifyArgs(args))
-	bt := &builtinSetVarSig{newBaseBuiltinFunc(args, ctx)}
-	bt.foldable = false
-	return bt.setSelf(bt), errors.Trace(err)
+func (c *setVarFunctionClass) getFunction(ctx context.Context, args []Expression) (sig builtinFunc, err error) {
+	if err = errors.Trace(c.verifyArgs(args)); err != nil {
+		return nil, err
+	}
+	bf := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString, tpString)
+	bf.foldable = false
```

Comment:
set `bf.tp.Flen`

Reply:
setVar would not return any value, 
so it may be not necessary to set Flen.

Reply:
if so, we can just `return "", false, nil` at the end of `func (b *builtinSetVarSig) evalString`

Reply:
still, we'd better set `bf.tp.Flen` in case this function is used in other context except for `set` statement